stdlib/experimental/influxdb/influxdb.go

```
@@ -0,0 +1,192 @@
+			q := make(neturl.Values, query.Len())
+			query.Range(func(k values.Value, v values.Value) {
+				q.Set(k.Str(), v.Str())
+			})
+			u.RawQuery = q.Encode()
+		}
+
+		if err := validator.Validate(u); err != nil {
+			return nil, err
+		}
+
+		url = u.String()
+	}
+
+	var req *http.Request
+	{
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, timeout.Duration())
+		defer cancel()
+
+		req, err = http.NewRequestWithContext(ctx, strings.ToUpper(method), url, bytes.NewReader(body))
+		if err != nil {
+			return nil, err
+		}
+
+		if headers != nil {
+			headers.Range(func(k values.Value, v values.Value) {
+				req.Header.Set(k.Str(), v.Str())
+			})
+		}
+		req.Header.Set("Authorization", "Token "+token)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		// Following the pattern set by the  experimental/http.get implementation:
+		// Alias the DNS lookup error so as not to disclose the DNS server address.
+		// This error is private in the net/http package, so string matching is used.
+		if strings.HasSuffix(err.Error(), "no such host") {
+			return nil, errors.New("no such host")
+		}
+		return nil, err
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	responseHeaders, err := headerToDict(resp.Header)
+	if err != nil {
+		return nil, err
+	}
+
+	return values.NewObjectWithValues(map[string]values.Value{
+		"statusCode": values.NewInt(int64(resp.StatusCode)),
+		"headers":    responseHeaders,
+		"body":       values.NewBytes(b)}), nil
+}
+
+// headerToDict constructs a values.Dictionary from a map of header keys and values.
+func headerToDict(header http.Header) (values.Dictionary, error) {
+	builder := values.NewDictBuilder(semantic.NewDictType(semantic.BasicString, semantic.BasicString))
+	for name, thevalues := range header {
+		for _, onevalue := range thevalues {
+			if err := builder.Insert(values.NewString(name), values.NewString(onevalue)); err != nil {
+				return nil, errors.New("")
```

Can you return the real error here instead of masking it behind an empty error?